app/src/main/java/com/fastlink/zekrapp/presentation/ZekirScreen.kt

```diff
@@ -0,0 +1,121 @@
+package com.fastlink.zekrapp.presentation
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.material.FabPosition
+import androidx.compose.material.Scaffold
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowForward
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalClipboardManager
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.fastlink.zekrapp.LocalNavController
+import com.fastlink.zekrapp.presentation.utils.AppBar
+import com.fastlink.zekrapp.presentation.utils.CustomFloatingActionButton
+import com.fastlink.zekrapp.presentation.utils.CustomSnackBar
+import com.fastlink.zekrapp.presentation.utils.ZekirCard
+import com.fastlink.zekrapp.presentation.utils.bottomAppBar.BottomBar
+import com.fastlink.zekrapp.presentation.utils.bottomAppBar.getListOfBottomBarItemsInZekirScreen
+import com.fastlink.zekrapp.viewModel.ZekirCategoryViewModel
+import com.fastlink.zekrapp.viewModel.ZekirViewModel
+import com.fastlink.zekrapp.viewModel.ZekirViewModelFactory
+
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun ZekirScreen(categoryId: Int) {
+    val categoryViewModel: ZekirCategoryViewModel = viewModel()
+    val zekirViewModel: ZekirViewModel = viewModel(factory = ZekirViewModelFactory(categoryId))
+    val snackBarState = remember { SnackbarHostState() }
+    val pagerState = rememberPagerState(
+        pageCount = { zekirViewModel.zekirs.size },
+    )
+    val scaffoldState = rememberScaffoldState()
+    val navController = LocalNavController.current
+    val clipboardManager = LocalClipboardManager.current
+    val zekirCategory = categoryViewModel.getCategoryById(categoryId)
```

Comment:
call this in the view model

Reply:
i didn't understand you ? what i do ?

Reply:
Call categoryViewModel.getCategoryById(categoryId) in the view model.

Reply:
i changed the name of function  to `getZeikrCategoryById` and  call function in  View Model ?

Reply:
@RoonyShekho09 
I did a mistake in this code i created an instance  in class ZekirCategpryViewModel 
```kotlin
 val categoryViewModel: ZekirCategoryViewModel = viewModel()
```  
but we don't need a new instance then i changed to this 
```kotlin
     val categoryViewModel = LocalViewModel.current
```

Reply:
As I said before, LocalComposition is not for everything. For now use viewModel() until you read about Hilt.

Reply:
i read abou it but i didn't understand it  inshallah today i will search for it

Reply:
This is what the documentation says:

> Avoid CompositionLocal for concepts that aren't thought as tree-scoped or sub-hierarchy scoped. A CompositionLocal makes sense when it can be potentially used by any descendant, not by a few of them.

Also:

> An example of a bad practice is creating a CompositionLocal that holds the ViewModel of a particular screen so that all composables in that screen can get a reference to the ViewModel to perform some logic. This is a bad practice because not all composables below a particular UI tree need to know about a ViewModel.